plugins/SimulationView/SimulationView.py

```diff
@@ -244,15 +243,20 @@ def advanceTime(self, time_increase: float) -> bool:
             self.setTime(self._current_time + time_increase)
         return True
 
-    def cumulativeLineDuration(self) -> Tuple[float, List[float]]:
-        # TODO: cache the total duration and cumulative line duration at each layer change event
-        cumulative_line_duration = []
-        total_duration = 0.0
-        for polyline in self.getLayerData().polygons:
-            for line_duration in list((polyline.lineLengths / polyline.lineFeedrates)[0]):
-                total_duration += line_duration
-                cumulative_line_duration.append(total_duration)
-        return total_duration, cumulative_line_duration
+    def cumulativeLineDuration(self) -> float:
+        # Make sure _cumulative_line_duration is initialized properly
+        if self.getCurrentLayer() not in self._cumulative_line_duration:
+            self._cumulative_line_duration[self.getCurrentLayer()] = []
+            total_duration = 0.0
+            for polyline in self.getLayerData().polygons:
+                for line_duration in list((polyline.lineLengths / polyline.lineFeedrates)[0]):
+                    total_duration += line_duration / SimulationView.SIMULATION_FACTOR
+                    self._cumulative_line_duration[self.getCurrentLayer()].append(total_duration)
+
+        # Calculate the total duration using numpy.sum
+        total_duration = (self._cumulative_line_duration[self.getCurrentLayer()][-1])
+
+        return total_duration
```

Comment:
Why return the `total_duration` in a function with the name `cumulativeLineDuration`

I would return the communative duration here
```suggestion
        return self._cumulative_line_duration[self.getCurrentLayer()]
```

And then where it is used also get the `total_duration`
```python
cumulative_line_duration = self.cumulativeLineDuration()
total_duration = cumulative_line_duration[-1]
```